sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -524,7 +533,9 @@ export class SessionReceiver {
     }
     this._createMessageSessionIfDoesntExist()
       .then(() => {
-        this._messageSession!.receive(onMessage, onError, options);
+        if (!this._isClosed) {
+          this._messageSession!.receive(onMessage, onError, options);
+        }
```

Comment:
I would also recommend adding an else block here where we call close on the messageSession.
Same for the non session case